sys/kern/main.c

```diff
@@ -5,9 +5,11 @@
 #include <sys/proc.h>
 #include <sys/thread.h>
 #include <sys/ktest.h>
+#include <sys/malloc.h>
 
 /* Borrowed from mips/malta.c */
 char *kenv_get(const char *key);
+char **kenv_get_init_args(int *n);
 
```

Comment:
I find it a little bit confusing that `kenv_get_init_args()` returns the result both as a regular return value and through a pointer. But maybe it's just nitpicking.

Reply:
From my point of view it is the simplest solution in this case. But if you know something more elegant (and not more complicated) feel free to share your idea.